extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/YoutubeParsingHelper.java

```diff
@@ -1419,49 +1405,44 @@ public static String getIosUserAgent(@Nullable final Localization localization)
     }
 
     /**
-     * Add required headers and cookies to an existing headers Map.
-     * @see #addClientInfoHeaders(Map)
-     * @see #addCookieHeader(Map)
+     * Returns a {@link Map} containing the required YouTube headers.
      */
-    public static void addYouTubeHeaders(final Map<String, List<String>> headers)
-            throws IOException, ExtractionException {
-        addClientInfoHeaders(headers);
-        addCookieHeader(headers);
+    public static Map<String, List<String>> getYouTubeHeaders()
+            throws ExtractionException, IOException {
+        final var headers = getClientInfoHeaders();
+        headers.put("Cookie", List.of(generateConsentCookie()));
```

Comment:
I think the downloader might need to add cookies, e.g. recaptcha cookies. Would it be able to do so with `List.of()`? I might be wrong on this though, since NewPipe itself [does not actually modify the headers](https://github.com/TeamNewPipe/NewPipe/blob/7aacaf8c38358ccd3da98fcb26ed9420d6fdd9ee/app/src/main/java/org/schabi/newpipe/DownloaderImpl.java#L145).

Reply:
> I might be wrong on this though, since NewPipe itself [does not actually modify the headers](https://github.com/TeamNewPipe/NewPipe/blob/7aacaf8c38358ccd3da98fcb26ed9420d6fdd9ee/app/src/main/java/org/schabi/newpipe/DownloaderImpl.java#L145).

You're likely right, that's how we do also it on Piped:

https://github.com/TeamPiped/Piped-Backend/blob/c2132f16dd5953cf9efbb36e4ac19778e589f855/src/main/java/me/kavin/piped/utils/DownloaderImpl.java#L44